topic-controller/src/test/java/io/strimzi/controller/topic/ControllerIntegrationTest.java

```diff
@@ -142,17 +143,19 @@ public void setup(TestContext context) throws Exception {
                 kafka = session.kafka;
                 k8s = session.k8s;
                 controller = session.controller;
-                topicsConfigWatcher = session.tcw;
-                topicsWatcher = session.tw;
+                topicsConfigWatcher = session.topicConfigsWatcher;
```

Comment:
I think there's already a `@Disabled` method for the partitions modification case in the `ControllerIntegrationTest`. In any case, we should add an integration test for this. It would be similar to the one for modifying topic configs, I guess.

Reply:
I added the test we needed.